### PR TITLE
Wired up Brave Wallet to NotifyUserInteraction

### PR DESCRIPTION
**Pull request by Douglashdaniel**
## Description 
Wired up the Brave Desktop Wallet to the `NotifyUserInteraction` method.

1) Created an event listener for mouse movements that starts when the wallet is unlocked and will stop after the wallet is locked.
2) Created an interval timer that will start once the wallet is unlocked that will trigger a check to see if the mouse has moved every 50 seconds, if the mouse has moved it will call the `NotifyUserInteraction` method to reset the timeout countdown.
If the wallet is locked, the interval will stop.

Note: (In the video, the interval is set to every 5 seconds for demo purposes.)

Resolves <https://github.com/brave/brave-browser/issues/18444>

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

https://user-images.githubusercontent.com/40611140/135361398-468bd476-bcd2-4990-b224-e78c991d2d57.mov

### CODE REVIEW

File: components/brave_wallet_ui/page/container.tsx [hunk 1]
```diff
@@ -110,6 +110,9 @@ function Container (props: Props) {
   const [buyAmount, setBuyAmount] = React.useState('')
   const [sendAmount, setSendAmount] = React.useState('')
   const [selectedWidgetTab, setSelectedWidgetTab] = React.useState<BuySendSwapTypes>('buy')
+  const [lastCheckedTime, setLastCheckedTime] = React.useState<number>()
+  const [currentMousePosition, setCurrentMousePosition] = React.useState({ x: 0, y: 0 })
+  const [lastMousePosition, setLastMousePosition] = React.useState({ x: 0, y: 0 })
```
Comment (onyb) [inline components/brave_wallet_ui/page/container.tsx:115]: Can we please move all this state and the associated `useMemo` / `useEffect` to a custom hook? This is not wallet-specific code so we could cut down the clutter a bit.

File: components/brave_wallet_ui/page/container.tsx [hunk 2]
```diff
@@ -146,6 +149,41 @@ function Container (props: Props) {
     swapQuote
   )
 
+  React.useMemo(() => {
+    const x = currentMousePosition.x
+    const y = currentMousePosition.y
```
Comment (onyb) [inline components/brave_wallet_ui/page/container.tsx:154]:
```suggestion
    const { x, y } = currentMousePosition
```

**Review by onyb — CHANGES_REQUESTED** [review]

File: components/brave_wallet_ui/page/container.tsx (hunk 1 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/page/container.tsx]: Done!

File: components/brave_wallet_ui/page/container.tsx (hunk 2 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/page/container.tsx]: Done!

**Review by onyb — APPROVED** [review]

**Pull request merged**